lib/iex/lib/iex/info.ex

```diff
@@ -35,13 +35,27 @@ defimpl IEx.Info, for: Atom do
       end
 
     mod_info = mod.module_info()
-    ["Module bytecode": module_object_file(mod),
-     "Source": module_source_file(mod_info),
-     "Version": module_version(mod_info),
-     "Compile options": module_compile_options(mod_info),
-     "Description": "#{extra}Call #{inspect mod}.module_info() to access metadata.",
-     "Raw representation": ":" <> inspect(Atom.to_string(mod)),
-     "Reference modules": "Module, Atom"]
+    generic_info =
+      ["Module bytecode": module_object_file(mod),
+       "Source": module_source_file(mod_info),
+       "Version": module_version(mod_info),
+       "Compile options": module_compile_options(mod_info),
+       "Description": "#{extra}Call #{inspect mod}.module_info() to access metadata."]
+
+    specific_info =
+      if _protocol? = function_exported?(mod, :__protocol__, 1) do
+        impls =
+          mod
+          |> Protocol.extract_impls(:code.get_path())
+          |> Enum.map_join(", ", &inspect/1)
+        ["Protocol": "This module is a protocol. These data structures implement it:\n  #{impls}"]
```

Comment:
@josevalim personally I prefer to relegate all the protocol info to `"Protocol"` for now but if someone else prefers your alternative let's go with it :)